pkgs/firehose/lib/src/repo.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:firehose/src/changelog.dart';
+import 'package:path/path.dart' as path;
+import 'package:yaml/yaml.dart' as yaml;
+
+import 'pubspec.dart';
+
+class Repo {
+  /// Returns true if this repository hosts only a single package, and that
+  /// package lives at the top level of the repo.
+  bool get singlePackageRepo {
```

Comment:
[nit] how about `isSinglePackage` or `containsSinglePackage`

https://dart.dev/guides/language/effective-dart/design#prefer-a-non-imperative-verb-phrase-for-a-boolean-property-or-variable